src/PointwiseFunctions/AnalyticData/GrMhd/CylindricalBlastWave.cpp

```diff
@@ -0,0 +1,228 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/AnalyticData/GrMhd/CylindricalBlastWave.hpp"
+
+#include <cmath>
+#include <pup.h>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataVector.hpp"  // IWYU pragma: keep
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Parallel/PupStlCpp11.hpp"
+#include "Utilities/ConstantExpressions.hpp"  // IWYU pragma: keep
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/Math.hpp" // IWYU pragma: keep
+
+namespace {
+template <typename DataType>
+Scalar<DataType> compute_piecewise(
+    const tnsr::I<DataType, 3, Frame::Inertial>& x, const double& inner_radius,
+    const double& outer_radius, const double& inner_value,
+    const double& outer_value) noexcept {
+  const DataType cylindrical_radius =
+      sqrt(square(get<0>(x)) + square(get<1>(x)));
+  auto piecewise_scalar = make_with_value<Scalar<DataType>>(x, 0.0);
+
+  // piecewise_scalar should equal inner_value for r < inner_radius,
+  // should equal outer_value for r > outer_radius, and should be a smooth
+  // transition in between. Here, use step_function() to compute the result
```

Comment:
I would not call the function below smooth.  It has kinks at the inner and outer radii.  I don't know whether the code or the comment should be changed.

Reply:
This wording comes from Sec. 6.2.3 of the first spectre paper, where this same transition is called smooth. The code can't change, because the transition function is part of a standard test. I've edited the comment in my next fixup commit to remove the word smooth.